via/static/scripts/video_player/components/VideoPlayerApp.tsx

```diff
@@ -118,6 +118,24 @@ export default function VideoPlayerApp({
     TranscriptData | APIError | null
   >(isTranscript(transcriptSource) ? transcriptSource : null);
 
+  // Notify client once the transcript is rendered, and ready to anchor
+  // annotations.
+  const [contentReady, resolveContentReady, rejectContentReady] =
+    useMemo(() => {
+      let reject: (err: Error) => void;
+      let resolve: () => void;
+      const promise = new Promise<void>((resolve_, reject_) => {
+        resolve = resolve_;
+        reject = reject_;
+      });
+      return [promise, resolve!, reject!];
```

Comment:
In future, dances like this can be simplified using [Promise.withResolvers](https://github.com/tc39/proposal-promise-with-resolvers).